### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by ngeelani48**
**In this PR @ngeelani48  and @Koshao-sys  fulfilled the following requirements:**

- Solved N+1 problem when fetching all posts and their comments for a user without using the bullet gem.

**Screenshot before**
Screenshot (99)

**Screenshot after**
Screenshot (102)

**Integration specs**

- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.
- User show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.
- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.
- Post-show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commenter.
  - I can see the comment each commenter left.

### CODE REVIEW

File: spec/integration/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    click_link(@post1.title)
+    expect(current_path).to eq(user_post_path(@user, @post1))
+  end
+
+  it 'redirects to the user posts index page when clicking on "See all posts"' do
+    visit user_path(@user)
+    click_link('See all posts')
+    expect(current_path).to eq(user_posts_path(@user))
+  end
+end
```
Comment (Whoistolu) [inline spec/integration/user_show_spec.rb]: - [ ] [REQUIRED] Kindly add tests that assert you can see a button that lets me view all of a user's posts as stated by the project requirements.

File: spec/integration/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    expect(page).to have_selector('.comments-container-index li', text: "#{@user.name}: I like it")
+    expect(page).to have_selector('.comments-container-index li', text: "#{@user.name}: Nice post")
+    expect(page).to have_selector('.comments-container-index li', text: 'No comments added yet...')
+  end
+
+  it 'redirects to a post show page when clicking on a post' do
+    visit user_posts_path(@user)
+    click_link('First Post')
+    expect(current_path).to eq(user_post_path(@user, @first_post))
+  end
```
Comment (Whoistolu) [inline spec/integration/post_index_spec.rb:54]: - [ ] [REQUIRED] Kindly add tests that assert you can see a section for pagination if there are more posts than fit on the view as stated in the project requirements.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @ngeelani48 and @Koshao-sys ,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers, and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi team,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/integration/post_index_spec.rb (hunk 2 above)
Comment (ngeelani48) [reply spec/integration/post_index_spec.rb:54]: @Whoistolu Thanks for the review and your time. I missed some of these tests.

**Pull request merged**